src/pages/index.js

```diff
@@ -34,29 +30,85 @@ const gallerySection = new Section(
   ".galery__cards"
 );
 
+modalsForms.forEach((form) => {
+  formvalidatorList[form.id] = new FormValidator(optionsValidation, form);
+  formvalidatorList[form.id].enableValidation();
+});
+
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__subtitle",
+  avatarSelector: ".profile__image",
 });
 
 const profileModal = new PopupWithForm(
   "#modal-profile",
   (formData) => {
-    userInfo.setUserInfo(formData.name, formData.job);
+    api
+      .editProfile({ name: formData.name, about: formData.job })
+      .then((responce) => {
+        userInfo.setUserInfo(responce.name, responce.about);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        profileModal.submissionCompleate();
+      });
   },
   formvalidatorList["profile"]
 );
+
 const cardsModal = new PopupWithForm(
   "#modal-cards",
   (formData) => {
-    gallerySection.addItem(
-      createCard({ name: formData.title, link: formData.link }),
-      true
-    );
+    api
+      .addCard({ name: formData.title, link: formData.link })
+      .then((response) => {
+        gallerySection.addItem(createCard(response), true);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        cardsModal.submissionCompleate();
+      });
   },
   formvalidatorList["cards"]
 );
+
+const avarEditModal = new PopupWithForm(
+  "#modal-edit-avatar",
+  (formData) => {
+    api
+      .editAvatar({ avatar: formData.avatar })
+      .then((responce) => {
+        userInfo.setAvatarLink(responce.avatar);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        avarEditModal.submissionCompleate();
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}

```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```